piff/optical_model.py

```diff
@@ -61,92 +98,123 @@ class Optical(Model):
                             it, defined to be positive in the counter-clockwise direction.
                             [default: 0. * galsim.degrees]
+    :param pad_factor       Numerical factor by which to pad the pupil plane FFT [default: 1]
+    :param oversampling     Numerical factor by which to oversample the FFT [default: 1]
     :param pupil_plane_im:  The name of a file containing the pupil plane image to use instead
                             of creating one from obscuration, struts, etc. [default: None]
 
-    Second, there may be an atmospheric component, which uses either a galsim.Kolmogorov or
-    galsim.VonKarman to model the profile.
+    The next two parameters are specific to our treatment of the Optical PSF. If given,
+    they are used to create an additional galsim.PhaseScreen describing the mirror's own contribution to the wavefront.
 
-    :param fwhm:            The full-width half-max of the atmospheric part of the PSF.
-                            [default: None]
-    :param r0:              The Fried parameter in units of meters to use for the Kolmogorov
-                            profile. [default: None]
-    :param L0:              The outer scale in units of meters if desired, in which case
-                            the atmospheric part will be a VonKarman. [default: None]
+    :param mirror_figure_im: The name of a file containing an additional phase contribution, for example
+                             from the figure of the primary mirror. [default: None]
+    :param mirror_figure_halfsize:   The radius of the image containting the mirror_figure_im. [default: None]
 
-    Finally, there is allowed to be a final Gaussian component and an applied shear.
 
-    :param sigma:           Convolve with gaussian of size sigma. [default: 0]
-    :param g1, g2:          Shear to apply to final image. Simulates vibrational modes.
-                            [default: 0]
+    Second, there is an atmospheric component, which uses either a galsim.Kolmogorov or
+    galsim.VonKarman to model the profile.
+
+    :param atmo_type        The name of the Atmospheric kernel. [default 'VonKarman']
+
+    Finally, there is both an additional Gaussian component to describe CCD diffusion and separately an applied shear.
+    :param sigma            Gaussian CCD diffusion in arcsec [default:0.]
 
     Since there are a lot of parameters here, we provide the option of setting many of them
     from a template value.  e.g. template = 'des' will use the values stored in the dict
-    piff.optical_model.optical_templates['des'].
+    piff.optatmo_model.aperture_templates['des'].
 
-    :param template:        A key word in the dict piff.optical_model.optical_template to use
-                            for setting values of these parameters.  [default: None]
+    :param template :       A key word in the dict piff.optatmo_model.aperture_template to use
```

Comment:
This doc is wrong.  It's `piff.optical_model.optical_templates` still.

Reply:
fixed

Reply:
There are still a couple places where aperture_template is used.

Reply:
this time for sure